README.md

```diff
@@ -20,21 +20,162 @@ Additionally `simdjson-go` has the following features:
+| TypeNone   | Nothing follows. Iter done |
+| TypeNull   | Null value                 |
+| TypeString | `String()`/`StringBytes()` |
+| TypeInt    | `Int()`/`Float()`          |
+| TypeUint   | `Uint()`/`Float()`         |
+| TypeFloat  | `Float()`                  |
+| TypeBool   | `Bool()`                   |
+| TypeObject | `Object()`                 |
+| TypeArray  | `Array()`                  |
+| TypeRoot   | `Root()`                   |
+
+The complex types returns helpers that will help parse each of the underlying structures.
+
+It is up to you to keep track of the nesting level you are operating at.
+
+
+## Parsing NDSJON stream
+
+Newline delimited json is sent as packets with each line being a root element.
+
+Here is an example that counts the number of `"Make": "HOND"` in NDSJON similar to this:
+
+```
+{"Age":20, "Make": "HOND"}
+{"Age":22, "Make": "TLSA"}
+```
+
+```Go
+func findHondas(r io.Reader) {
+	// Temp values.
+	var tmpO simdjson.Object{}
+	var tmpE simdjson.Element{}
+	var tmpI simdjson.Iter
+	var nFound int
+	
+	// Communication
+	reuse := make(chan *simdjson.ParsedJson, 10)
+	res := make(chan simdjson.Stream, 10)
+
+	simdjson.ParseNDStream(r, res, reuse)
+	// Read results in blocks...
+	for got := range res {
+		if got.Error != nil {
+			if got.Error == io.EOF {
+				break
+			}
+			log.Fatal(got.Error)
+		}
+
+		all := got.Value.Iter()
+		// NDJSON is a separated by root objects.
+		for all.Advance() == simdjson.TypeRoot {
+			// Read inside root.
+			t, i, err := all.Root(&tmpI)
+			if t != simdjson.TypeObject {
+				log.Println("got type", t.String())
+				continue
+			}
+
+			// Prepare object.
+			obj, err := i.Object(&tmpO)
+			if err != nil {
+				log.Println("got err", err)
+				continue
+			}
+
+			// Find Make key.
+			elem := obj.FindKey("Make", &tmpE)
+			if elem.Type != TypeString {
+				log.Println("got type", err)
+				continue
+			}
+			
+			// Get value as bytes.
+			asB, err := elem.Iter.StringBytes()
+			if err != nil {
+				log.Println("got err", err)
+				continue
+			}
+			if bytes.Equal(asB, []byte("HOND")) {
+				nFound++
+			}
+		}
+		reuse <- got.Value
+	}
+	fmt.Println("Found", nFound, "Hondas")
+}
+```
+
+More examples can be found in the examples subdirectory and further documentation can be found at [godoc](https://pkg.go.dev/github.com/minio/simdjson-go?tab=doc). 
+
+
 ## Performance vs simdjson
 
 Based on the same set of JSON test files, the graph below shows a comparison between `simdjson` and `simdjson-go`.
 
 ![simdjson-vs-go-comparison](chart/simdjson-vs-simdjson-go.png)
 
-These numbers were measured on a MacBook Pro equipped with a 3.1 GHz Intel Core i7. 
-Also, to make it a fair comparison, the constant `GOLANG_NUMBER_PARSING` was set to `false` (default is `true`) 
+These numbers were measured on a MacBook Pro equipped with a 3.1 GHz Intel Core i7.
+Also, to make it a fair comparison, the constant `GOLANG_NUMBER_PARSING` was set to `false` (default is `true`)
```

Comment:
This remark is now no longer applicable

Reply:
Well, the numbers aren't updated, since I still only have AVX2 available, but it should be fair to expect that we are pretty close to the numbers.